--- gettext-runtime/configure.ac.orig	2012-08-09 14:29:56.805783472 +0100
+++ gettext-runtime/configure.ac	2012-08-09 14:30:45.122393305 +0100
@@ -30,6 +30,7 @@
 
 dnl Checks for programs.
 AC_PROG_CC
+AC_PROG_CXX
 AC_PROG_INSTALL
 AC_PROG_YACC
 
@@ -114,9 +115,6 @@
 CROSS_COMPILING=$cross_compiling
 AC_SUBST([CROSS_COMPILING])
 
-dnl Checks for optional programs for the tests/lang-* tests.
-gt_PROG_ANSI_CXX
-
 dnl Checks for special libraries for the tests/test-lock test.
 dnl On some systems, sched_yield is in librt, rather than in libpthread.
 LIBSCHED=
