--- m4/hooks.m4.orig	2012-08-12 18:44:20.078029129 +0100
+++ m4/hooks.m4	2012-08-12 18:47:27.993458415 +0100
@@ -85,14 +85,16 @@
   AC_MSG_CHECKING([whether to use nettle])
 if test "$cryptolib" = "nettle";then
   AC_MSG_RESULT(yes)
-    AC_LIB_HAVE_LINKFLAGS([nettle], [hogweed gmp], [#include <nettle/rsa.h>],
-                          [rsa_decrypt_tr (0,0,0,0,0,0,0)])
-    if test "$ac_cv_libnettle" != yes; then
-      AC_MSG_ERROR([[
+    AC_CHECK_LIB([nettle], [nettle_rsa_decrypt_tr], [ac_cv_libnettle=yes],
+    	AC_MSG_ERROR([[
   *** 
   *** Libnettle 2.5 was not found. Note that you must compile nettle with gmp support.
-  ]])
-    fi
+	]]), [-lhogweed -lgmp])
+  AC_SUBST([HAVE_LIBNETTLE])
+  AC_SUBST([LTLIB_NETTLE])
+  HAVE_LIBNETTLE=1
+  LTLIB_NETTLE="-lnettle -lhogweed -lgmp"
+  LIBS="$LIBS -lnettle -lhogweed -lgmp"
 else
   AC_MSG_RESULT(no)
 fi
