--- inc/Module/Install/PRIVATE/Net/SSLeay.pm.orig	2012-08-09 12:47:01.889929160 +0100
+++ inc/Module/Install/PRIVATE/Net/SSLeay.pm	2012-08-09 12:48:51.695276411 +0100
@@ -204,24 +204,26 @@
 
 sub check_openssl_version {
     my ($self, $prefix, $exec) = @_;
-    my ($major, $minor, $letter);
+    my ($output, $major, $minor, $letter);
 
-    {
+    if (defined $ENV{OPENSSL_VERSION}) {
+        $output = $ENV{OPENSSL_VERSION}
+    } else {
         my $pipe = gensym();
         open($pipe, "$exec version |")
             or die "Could not execute $exec";
-        my $output = <$pipe>;
+        $output = <$pipe>;
         chomp $output;
         close $pipe;
+    }
 
-        unless ( ($major, $minor, $letter) = $output =~ /^OpenSSL\s+(\d+\.\d+)\.(\d+)([a-z]?)/ ) {
+    unless ( ($major, $minor, $letter) = $output =~ /^OpenSSL\s+(\d+\.\d+)\.(\d+)([a-z]?)/ ) {
             die <<EOM
 *** OpenSSL version test failed
     (`$output' has been returned)
     Either you have bogus OpenSSL or a new version has changed the version
     number format. Please inform the authors!
 EOM
-        }
     }
 
     print "*** Found OpenSSL-${major}.${minor}${letter} installed in $prefix\n";
